### PR TITLE
chore: migrate to latest UnJS project template

### PR DESCRIPTION
**Pull request by johannschopplich**
- Migrate to pnpm
- Replace Jest with vitest
- Use unbuild
- Add missing types

## Todo if Merging

- [x] Rename `master` to `main`

### CODE REVIEW

File: LINCENSE
Comment (Barbapapazes) [inline LINCENSE:1]: There is a typo in the filename

File: README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@
 Install:
 
 ```bash
-yarn add theme-colors
```
Comment (Barbapapazes) [inline README.md:14]: I think we could update using the installation section in the template part. https://github.com/unjs/template?tab=readme-ov-file#usage

File: package.json [hunk 2]
```diff
@@ -1,34 +1,45 @@
...
     "colors",
     "shades",
     "tailwind",
     "tailwindcss"
   ],
-  "repository": "nuxt-contrib/theme-colors",
-  "license": "MIT",
+  "sideEffects": false,
+  "type": "module",
+  "exports": {
```
Comment (Barbapapazes) [inline package.json:16]: Could you update this part of the package?

Here an exemple:
```json
"exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.mts",
        "default": "./dist/index.mjs"
      },
      "require": {
        "types": "./dist/index.d.cts",
        "default": "./dist/index.cjs"
      }
    }
  },
  "main": "./dist/index.cjs",
  "types": "./dist/index.d.ts",
```

**Review by Barbapapazes — CHANGES_REQUESTED** [review]

File: LINCENSE
Comment (johannschopplich) [reply LINCENSE:1]: Good catch! Eagle eye. 🦅 👁️

File: package.json (hunk 2 above)
Comment (johannschopplich) [reply package.json:16]: Sure. I have update to the latest UnJS repo template, which slightly differs. But I guess to prefer the template is better than customizing it (although I might prefer the syntax you posted).

**Comment by johannschopplich** [comment]
@Barbapapazes Thanks for the input! This was my first contribution to UnJS. I forgot it even existed. 😄 
I have migrated to the latest UnJS template. New PR is: https://github.com/unjs/theme-colors/pull/30.

**Pull request closed without merging**